examples/counter_functional/src/main.rs

```diff
@@ -0,0 +1,28 @@
+use yew::prelude::*;
+
+#[function_component(App)]
+fn app() -> Html {
+    let state = use_state(|| 0);
+
+    let incr_counter = {
+        let state = state.clone();
+        Callback::from(move |_| state.set(*state + 1))
```

Comment:
I think these could be improved with the `use_callback` hooks. As is, the callback changes identity everytime the `App` rerenders, which is not wrong, but re-registers the `onclick` below on every render.

Reply:
Try using `state` instead of `()` as dependency.

Reply:
Oh right. If you clone it a second time, you could just pass `state2`. Probably `use_memo` and `use_callback` need a little bit more tuning to make them user friendly in that regard. Let's not do that for now, though.